lib/util/string_conv/string_conv.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef STRING_CONV_H__
+#define STRING_CONV_H__
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Convert string to long param.
+ *
+ * @note On failure the passed value reference will not be altered.
+ *
+ * @param str Input string
+ * @param val Converted value
+ *
+ * @return 0 on success, negative error code on failure
+ */
+int string_conv_str2long(char *str, int *val);
```

Comment:
What is proper string for conversion? Can it start and end with white characters?
Is `   -10   ` proper number?

Reply:
In its current state the library does not trim heading or trailing whitespace. Strings with whitespace will be rejected.